cmd/harmony/main.go

```diff
@@ -311,7 +360,11 @@ func setupConsensusAndNode(nodeConfig *nodeconfig.ConfigType) *node.Node {
 	currentConsensus, err := consensus.New(
 		myHost, nodeConfig.ShardID, p2p.Peer{}, nodeConfig.ConsensusPriKey, decider,
 	)
-	currentConsensus.SelfAddress = common.ParseAddr(initialAccount.Address)
+
+	currentConsensus.SelfAddress = make(map[string]ethCommon.Address)
+	for _, initialAccount := range initialAccounts {
```

Comment:
leave def of selfaddress alone, instead add new field of  a map[shard.BlsPublicKey]struct{}

Reply:
why do we need two separate fields? The only use of the SelfAddress is to correctly assign the address of the block proposer while proposing the block. 
`coinbase := node.Consensus.SelfAddress[node.Consensus.LeaderPubKey.SerializeToHexStr()]` 
SelfAddress is not used anywhere else.

Reply:
To lessen the pain of merge conflicts to master